sql/src/main/java/io/crate/operation/scalar/ConcatFunction.java

```diff
@@ -131,6 +133,27 @@ public BytesRef evaluate(Input[] args) {
                 return new StringConcatFunction(new FunctionInfo(new FunctionIdent(NAME, dataTypes), DataTypes.STRING));
             } else {
                 for (int i = 0; i < dataTypes.size(); i++) {
+
+                    if(dataTypes.get(i) instanceof ArrayType){
```

Comment:
don't check for array types inside the loop, use another ``else if`` branch like the one producing the ``StringConcatFunction`` above